sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -83,17 +84,18 @@ def __init__(
         self.id = id
         self.container_link = "{}/colls/{}".format(database_link, self.id)
         self.client_connection = client_connection
-        self._properties = properties
         self._is_system_key: Optional[bool] = None
         self._scripts: Optional[ScriptsProxy] = None
+        if properties:
+            self.client_connection.container_properties_cache[self.container_link] = _set_properties_cache(properties)
 
     def __repr__(self) -> str:
         return "<ContainerProxy [{}]>".format(self.container_link)[:1024]
 
     def _get_properties(self) -> Dict[str, Any]:
-        if self._properties is None:
-            self._properties = self.read()
-        return self._properties
+        if self.container_link not in self.client_connection.container_properties_cache:
+            self.read()
+        return self.client_connection.container_properties_cache[self.container_link]
```

Comment:
same question: _get_properties should always translate into self.read?

Reply:
That is only when we first populate the cache, in the container itself the properties we are most actively using are rid, partition key definition, and container link, since those don't change, it saves resources by caching them on the first container read which might happen before _get_properties is called, in which case we don't need to do another read. Even if we do cache all the properties, we would just end up needing to do a read anyways for the other ones.